File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.java
class Solution {
    public int solution(String[][] book_time) {
        int answer = 0;
        int [] oneDayTimes = new int [1440];
        
            for( String [] book : book_time){
                int start = toTime(book[0]);
                int end =  Math.min(1439,toTime(book[1])+10);
                for( int a=start; a<end; a++){
                    oneDayTimes[a] ++;
                }
            }
        
            for(int a=0;a<oneDayTimes.length;a++){
                answer = Math.max(answer, oneDayTimes[a]);
            }       
            
            return answer;
    }
    public int toTime (String hourTime){
        String[] minTime = hourTime.split(":");
        
        return Integer.parseInt(minTime[0]) * 60 + Integer.parseInt(minTime[1]);
    }
}

File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.js
function solution(book_time) {
    var answer = 0;
    const oneDayTimes = [];
    for(let c = 0; c<1440;c++){
        oneDayTimes.push(0)
    }
    for(let book of book_time){
        const  start = toTime(book[0]);
        const  end =  Math.min(1439,toTime(book[1])+10);
        for( let a=start; a<end; a++){
            oneDayTimes[a] ++;
        }
    }
    for(let b=0;b<oneDayTimes.length;b++){
        answer = Math.max(answer, oneDayTimes[b]);
    }       
    return answer;
}

function toTime ( hourTime){
    const minTime = hourTime.split(":");
    const time = parseInt(minTime[0]) * 60 + parseInt(minTime[1]);
    return time;
}